lib/pure/oids.nim

```diff
@@ -66,21 +67,22 @@ proc oidToString*(oid: Oid, str: cstring) =
   str[24] = '\0'
 
 proc `$`*(oid: Oid): string =
+  ## Converts an oid to string.
   result = newString(24)
   oidToString(oid, result)
 
-proc rand(): cint {.importc: "rand", header: "<stdlib.h>", nodecl.}
-proc srand(seed: cint) {.importc: "srand", header: "<stdlib.h>", nodecl.}
-
-var t = getTime().toUnix.int32
-srand(t)
 
 var
-  incr: int = rand()
-  fuzz: int32 = rand()
+  t = getTime().toUnix.int32
+  seed = initRand(t)
```

Comment:
much better (at least doesn't impact std/random) but still has a flaw:
```nim
when true:
  import std/oids
  proc main()=
    let t = genOid()
    echo t
  main()
```

nim c main
./main
./main # if within 1 second, gives same results

fix: see how `randomize` works from full resolution time.
ideally should be factored (but privateImport isn't accepted yet... sigh...) so in the meantime all that's left is std/private/xxx or duplicating that (admitedly small) code snippet

Reply:
The issue exists before. Not the scope of this PR.

Reply:
indeed; => https://github.com/timotheecour/Nim/issues/429